grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/GRpcServerRunner.java

```diff
@@ -150,14 +153,15 @@ public void destroy() throws Exception {
             s.shutdown();
             int shutdownGrace = gRpcServerProperties.getShutdownGrace();
             try {
-                // If shutdownGrace is 0, then don't call awaitTermination
-                if (shutdownGrace < 0) {
+                if (shutdownGrace < 1) {
                     s.awaitTermination();
-                } else if (shutdownGrace > 0) {
+                } else {
```

Comment:
The original intent was to preserve existing behavior by avoiding a call to `awaitTermination` if shutdownGrace is 0 (the default value). I don't suspect that the extra call will be problematic, but just wanted to point it out

Reply:
The existing behavior is (was) to call awaitTermination() , but in awaiter thread. It does make sense to call this function in destroy method. So to preserve this behavior we set grace period to 0. In future I will implement smart beam lifecycle to allow user to start/stop the server manually